static/js/chart/draw.ts

```diff
@@ -629,46 +625,54 @@ function drawGroupLineChart(
  * @param id: This is the id for the chart legend element.
  * @param plotParams: It contains all colors and dashes for geoIds and statVars.
  */
-function buildInChartLegend(id: string, plotParams: PlotParams) {
-  const legend = d3.select("#" + id);
-  if (Object.keys(plotParams.dashes).length === 1) {
+function buildInChartLegend(
+  legend: d3.Selection<SVGGElement, any, any, any>,
+  plotParams: PlotParams,
+  dashInLegend: boolean,
+  legenTextdWidth: number
+) {
+  let yOffset = 0;
+  if (!dashInLegend) {
     // Only have one geoId. Then different statsVars should have different colors.
     const statsVars = Object.keys(plotParams.colors);
-    for (let i = 0; i < statsVars.length; i++) {
-      legend
-        .append("g")
-        .attr("transform", `translate(0, ${LEGEND.height * i})`)
+    for (const statsVar of statsVars) {
+      const text = legend
         .append("text")
-        .attr("x", "40")
-        .attr("y", `${LEGEND.height * i}`)
-        .text(plotParams.title[statsVars[i]])
+        .attr("transform", `translate(0, ${yOffset})`)
+        .attr("text-anchor", "start")
+        .attr("dy", ".3em")
+        .attr("fill", `${plotParams.colors[statsVar]}`)
         .style("font-size", "14")
-        .attr("fill", `${plotParams.colors[statsVars[i]]}`);
+        .text(plotParams.title[statsVar])
+        .call(wrap, legenTextdWidth);
+      yOffset += text.node().getBBox().height + LEGEND.lineMargin;
```

Comment:
we should be careful with getbbox, since it wont work if the chart isn't attached to the dom. any issues on the place page?

Reply:
done

Reply:
yeah, this works fine; and this is only used in gni page. I wonder if we need to do the same thing for landing page as now there is some text cut.